packages/endo/src/import.js

```diff
@@ -0,0 +1,58 @@
+/* global StaticModuleRecord */
+/* eslint no-shadow: 0 */
+
+import { compartmentMapForNodeModules } from "./compartmap.js";
+import { search } from "./search.js";
+import { assemble } from "./assemble.js";
+
+const decoder = new TextDecoder();
+
+const resolve = (rel, abs) => new URL(rel, abs).toString();
+
+const makeImportHookMaker = (read, rootLocation) => packageLocation => {
```

Comment:
ugh fancy ML-inspired higher-order functions :). This would be easier for my not-so-clever brain to read if there was a `makeImportHook` identifier declared somewhere. I shall endeavour to mentally translate this into

```js
function makeImportHookMaker(read, rootLocation) {
  function makeImportHook(packageLocation) {
   ...
  }
  return makeImportHook;
}
```

Reply:
Oh in fact there are three levels of nesting here? The function defined+returned on line 14 is the `importHook` itself? I missed that the first time through, which either says something about my reading ability or the readability of nested fat arrow functions.

Reply:
Fair!